Support second-level precision for `assert_enqueued_with`

stevepolitodesign opened the pull request:
### Motivation / Background

Closes #52478

Prior to this commit, tests using [assert_enqueued_with][] asserting against the `at` argument would incorrectly pass if testing with values that used second-level precision.

```ruby
class BuggyJobTest < ActiveJob::TestCase
  def test_should_not_find_job
    at =  Time.new("1989-12-31T12:00:01")
    wait_until = Time.new("1989-12-31T12:00:02")

    assert_enqueued_with(job: BuggyJob, at:) do
      BuggyJob.set(wait_until:).perform_later
    end

    fail "Should not reach this line because assert_enqueued_with should have failed"
  end
end
```

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

[assert_enqueued_with]: https://edgeapi.rubyonrails.org/classes/ActiveJob/TestHelper.html#method-i-assert_enqueued_with

stevepolitodesign commented on activejob/lib/active_job/test_helper.rb:
```diff
@@ -736,7 +736,7 @@ def prepare_args_for_assertion(args)
 
           if arguments[:at].acts_like?(:time)
-            at_range = arguments[:at] - 1..arguments[:at] + 1
+            at_range = arguments[:at] - 1.second..arguments[:at] + 1.second
```
Since there's no `assert_not_enqueued_with` helper, I wasn't sure the best way to test this. One alternative would be to bypass the `assert_enqueued_with` altogether, and manually look at the value set.

seanpdoyle replied:
It would be valuable to continue to explore adding test coverage. 

Either way, a change like this warrants adding an entry to [activejob/CHANGELOG.md](https://github.com/rails/rails/blob/main/activejob/CHANGELOG.md).

byroot commented:
The CI failure is legit. Also please squash your commits.

stevepolitodesign committed:
Support second-level precision for `assert_enqueued_with`

Closes #52478

Prior to this commit, tests using [assert_enqueued_with][] asserting
against the `at` argument would incorrectly pass if testing with values
that used second-level precision.

```ruby
class BuggyJobTest < ActiveJob::TestCase
  def test_should_fail
    at =  Time.new("1989-12-31T12:00:01")
    wait_until = Time.new("1989-12-31T12:00:02")

    assert_not_equal at, wait_until
    assert_enqueued_with(job: BuggyJob, at:) do
      BuggyJob.set(wait_until:).perform_later
    end
  end
end
```

[assert_enqueued_with]: https://edgeapi.rubyonrails.org/classes/ActiveJob/TestHelper.html#method-i-assert_enqueued_with
